### ISSUE TITLE
I ask a question because of the low mAP score.

### ISSUE THREAD

**Issue by moonplay85** [dbolya/yolact#539]
I am using this code.
I have a few questions.
I am doing detection using a custom dataset.
However, using the guide, I set the data set path and gave commands.
The problem is that mAP levels are not coming up.
As you can see, the box and mask values are good.
Object detection is also good with the model created there.
The problem is that the mAP score doesn't go up.
Is there any way to solve this problem?
![2](https://user-images.githubusercontent.com/56929748/94504611-282cfa00-0244-11eb-921a-dcb256d41df6.jpg)

**Comment by areebsyed** [comment]
@mmiisskk how to change epochs for training? I cant find the line where I can denote the number of iterations I want to do training for

**Comment by jinfagang** [comment]
I think `achor` is really matters since yolact based on retinanet architecture as detector.

However, this is the bottleneck of yolact. the detector actually peformance not very good even you have a perfect anchor.

**Comment by areebsyed** [comment]
@jinfagang could you please tell me how to change number of epochs/training?

**Comment by mmiisskk** [comment]
> @mmiisskk how to change epochs for training? I cant find the line where I can denote the number of iterations I want to do training for

Epoch setting isn't an argument, it is computed in the beggining. It depends on dataset, max iteration set in cfg file and batch size.

Computations is in train file, you can change it as you need:
267   epoch_size = len(dataset) // args.batch_size
268   num_epochs = math.ceil(cfg.max_iter / epoch_size)

**Comment by areebsyed** [comment]
@mmiisskk thankyou so much. I just have a weird question, someone was asking me whats the output of this network, a polygon? I didnt understand the question. I think he doesnt mean a resulting mask because thats obvious from the output images. Could you tell me the answer to this.

**Comment by mmiisskk** [comment]
@areebsyed 
The output is list of tensors for detected objects. It contains classes, scores, boxes and mask for evaluated image.
The eval.py file return recognized image with all these informations add in it. 
Recognition is save in 'preds' in evalimg functions, and postprocessing of predict result is in the prep_display function (row 136) - you can check how it works, if you want.
Is this answer enough for you?

**Comment by areebsyed** [comment]
@mmiisskk thankyou very much for nice explanation. I tried to understand what you mean by analysing the preds and prep_display function. What I understood vaguely from the code and due to my lack of knowledge is that in preds it has the information on the coordinates of the polygon in form of tensors. Maybe what I am saying is wrong. My question is simple can I extract those mask polygon coordinates from the code. Or It is just the bbox coordinates that I can extract?
